lib/server/socket.js

```diff
@@ -25,6 +25,10 @@ module.exports = ({ server, filters, config }) => {
           return connections.set(token, clientPool);
         }
         debug('removing client %s', token);
+        // TODO: log/emit warning
+        connections.set(
+          'disconnected',
+          (connections.get('disconnected') || new Set()).add(token));
```

Comment:
I don't think this should be tracked in the same `Map` that holds actual sockets – too much potential for confusion.